packages/sdk/src/browser-wallets/oyl/index.ts

```diff
@@ -0,0 +1,150 @@
+import { Psbt } from "bitcoinjs-lib";
+
+import {
+  BrowserWalletNetworkMismatchError,
+  BrowserWalletNotInstalledError,
+  getAddressFormat,
+} from "../..";
+import { BrowserWalletNetwork } from "../../config/types";
+import {
+  BrowserWalletExtractTxFromNonFinalizedPsbtError,
+  OrditSDKError,
+} from "../../errors";
+import { BrowserWalletSignResponse, WalletAddress } from "../types";
+import { OylSignPSBTOptions } from "./types";
+
+function validateExtension(network: BrowserWalletNetwork = "mainnet"): void {
+  if (!isInstalled()) {
+    throw new BrowserWalletNotInstalledError("Oyl Wallet not installed");
+  }
+
+  if (network !== "mainnet") {
+    throw new BrowserWalletNetworkMismatchError(
+      "Oyl Wallet only supports mainnet",
+    );
+  }
+}
```

Comment:
could you move this function declaration down?

Reply:
updated